cmd/logviewer/main.go

```diff
@@ -0,0 +1,261 @@
+	cliApp.Version = fmt.Sprintf("%s/%s/%s-%s", "1.0.0", runtime.Version(), runtime.GOOS, runtime.GOARCH)
+	cliApp.Usage = "This is the entry point for starting a new Elrond node - the app will start after the genesis timestamp"
+	cliApp.Flags = []cli.Flag{
+		address,
+		logLevelPatterns,
+		logFile,
+		testServer,
+	}
+	cliApp.Authors = []cli.Author{
+		{
+			Name:  "The Elrond Team",
+			Email: "contact@elrond.com",
+		},
+	}
+}
+
+func startLogViewer(ctx *cli.Context) error {
+	var err error
+	logLevelPattern := ctx.GlobalString(logLevelPatterns.Name)
+	logLevels, _, err := logger.ParseLogLevelAndMatchingString(logLevelPattern)
+	if err != nil {
+		return err
+	}
+
+	if ctx.GlobalBool(testServer.Name) {
+		_ = testing.StartTestWebSocket("127.0.0.1:8080")
+		time.Sleep(time.Second * 2)
+	}
+
+	if ctx.GlobalBool(logFile.Name) {
+		err = prepareLogFile()
+		if err != nil {
+			return err
+		}
+
+		defer func() {
+			_ = fileForLogs.Close()
+		}()
+	}
+
+	addrString := ctx.GlobalString(address.Name)
+	webSocket, err = openWebSocket(addrString, logLevelPattern)
+	if err != nil {
+		return err
+	}
+	go listeningOnWebSocket()
+
+	//set this log's level to the lowest desired log level that matches received logs from elrond-go
+	lowestLogLevel := getLowestLogLevel(logLevels)
+	log.SetLevel(lowestLogLevel)
+
+	waitForUserToTerminateApp(webSocket)
+
+	return nil
+}
+
+func getLowestLogLevel(logLevels []logger.LogLevel) logger.LogLevel {
+	lowest := logLevels[0]
+	for i := 1; i < len(logLevels); i++ {
+		if lowest > logLevels[i] {
+			lowest = logLevels[i]
+		}
+	}
+
+	return lowest
+}
+
+func prepareLogFile() error {
+	var workingDir = ""
+	workingDir, err := os.Getwd()
+	if err != nil {
+		log.LogIfError(err)
+		workingDir = ""
+	}
+
+	logDirectory := filepath.Join(workingDir, defaultLogPath)
+	fileForLogs, err = core.CreateFile("logviewer", logDirectory, "log")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func openWebSocket(address string, logLevelPatterns string) (*websocket.Conn, error) {
+	u := url.URL{
+		Scheme: "ws",
+		Host:   address,
+		Path:   "/log",
+	}
+	conn, _, err := websocket.DefaultDialer.Dial(u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	err = conn.WriteMessage(websocket.TextMessage, []byte(logLevelPatterns))
+	if err != nil {
+		return nil, err
+	}
+
+	return conn, nil
+}
+
+func listeningOnWebSocket() {
+	for {
+		msgType, message, err := webSocket.ReadMessage()
+		if msgType == websocket.CloseMessage {
+			return
+		}
+		if err != nil {
+			isConnectionClosed := strings.Contains(err.Error(), "websocket: close")
+			if !isConnectionClosed {
+				log.Error("logviewer websocket error", "error", err.Error())
+				manualStop <- struct{}{}
+
+			}
+			return
+		}
+
+		outputMessage(message)
+	}
+}
+
+func waitForUserToTerminateApp(conn *websocket.Conn) {
+	stop := make(chan struct{}, 1)
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
```

Comment:
why the dedicate routine to just forward the event and not change the first case to receive from sigs?

Reply:
Right, removed go routine and stop chan